pkg/objstore/objstore.go

```diff
@@ -133,21 +137,33 @@ func DownloadFile(ctx context.Context, bkt BucketReader, src, dst string) error
 }
 
 // DownloadDir downloads all object found in the directory into the local directory.
-func DownloadDir(ctx context.Context, bkt BucketReader, src, dst string) error {
+func DownloadDir(ctx context.Context, logger log.Logger, bkt BucketReader, src, dst string) error {
 	if err := os.MkdirAll(dst, 0777); err != nil {
 		return errors.Wrap(err, "create dir")
 	}
-	err := bkt.Iter(ctx, src, func(name string) error {
+
+	var downloadedFiles []string
+	if err := bkt.Iter(ctx, src, func(name string) error {
 		if strings.HasSuffix(name, DirDelim) {
-			return DownloadDir(ctx, bkt, name, filepath.Join(dst, filepath.Base(name)))
+			return DownloadDir(ctx, logger, bkt, name, filepath.Join(dst, filepath.Base(name)))
```

Comment:
If `DownloadDir` fails, it should delete all files but you may end up with some empty directories. Is that ok?